pkg/didcomm/protocol/issuecredential/states.go

```diff
@@ -262,8 +267,15 @@ func (s *proposalSent) CanTransitionTo(st state) bool {
 	return st.Name() == stateNameOfferReceived || st.Name() == stateNameAbandoning
 }
 
-func (s *proposalSent) ExecuteInbound(_ *metaData) (state, stateAction, error) {
-	return nil, nil, fmt.Errorf("%s: ExecuteInbound is not implemented yet", s.Name())
+func (s *proposalSent) ExecuteInbound(md *metaData) (state, stateAction, error) {
+	// creates the state's action
+	action := func(messenger service.Messenger) error {
+		// sets message type
+		md.proposeCredential.Type = ProposeCredentialMsgType
```

Comment:
Are we replying to a proposal with another proposal?

Reply:
No, when we received an offer we have three ways:
1. Send a request credential (action is not triggered bc data is correct)
2. Send a propose credential  (e.g Continue())
3. Go to the abandoning state (e.g Stop())

This code is related to the `3` option. This means that we received an offer and data is not correct and we want to continue. This line shows that we are going back from `offerReceived` to `proposalSent`
https://github.com/hyperledger/aries-framework-go/pull/1408/files#diff-db914b8de9137376b5b65d627b223fdfR305